test/cloudtest/test_recovery.py

```diff
@@ -0,0 +1,252 @@
+
+    def get_mv_name(self) -> str:
+        return f"mv_{self.testcase_index}"
+
+
+def populate(mz: MaterializeApplication, cluster_setup: ClusterSetup) -> None:
+    mz.environmentd.sql(
+        "ALTER SYSTEM SET enable_managed_clusters = true",
+        port="internal",
+        user="mz_system",
+    )
+
+    mz.environmentd.sql(
+        f"CREATE CLUSTER {cluster_setup.get_storage_name()} SIZE = '1', REPLICATION FACTOR {cluster_setup.storage_replication_factor}",
+        port="internal",
+        user="mz_system",
+    )
+
+    mz.environmentd.sql(
+        f"CREATE CLUSTER {cluster_setup.get_compute_name()} SIZE = '1', REPLICATION FACTOR {cluster_setup.compute_replication_factor}",
+        port="internal",
+        user="mz_system",
+    )
+
+    counter_source_name = cluster_setup.get_counter_source_name()
+    mv_name = cluster_setup.get_mv_name()
+
+    mz.testdrive.run(
+        input=dedent(
+            f"""
+            > SELECT name FROM mz_clusters WHERE name IN ('{cluster_setup.get_storage_name()}', '{cluster_setup.get_compute_name()}');
+            {cluster_setup.get_storage_name()}
+            {cluster_setup.get_compute_name()}
+
+            > CREATE SOURCE {counter_source_name} IN CLUSTER {cluster_setup.get_storage_name()}
+              FROM LOAD GENERATOR COUNTER
+              (TICK INTERVAL '500ms');
+
+            > SELECT COUNT(*) FROM (SHOW SOURCES) WHERE name = '{counter_source_name}';
+            1
+
+            > CREATE MATERIALIZED VIEW {mv_name} (f1) IN CLUSTER {cluster_setup.get_compute_name()} AS SELECT * FROM {counter_source_name};
+
+            > CREATE DEFAULT INDEX IN CLUSTER {cluster_setup.get_compute_name()} ON {mv_name};
+
+            > SELECT COUNT(*) > 0 from {mv_name};
+            true
+            """
+        ),
+        no_reset=True,
+    )
+
+
+def validate_state(
+    mz: MaterializeApplication,
+    cluster_setup: ClusterSetup,
+    reached_index: int,
+    must_exceed_reached_index: bool,
+    timeout_in_sec: int,
+) -> None:
+    comparison_operator = ">" if must_exceed_reached_index else ">="
+
+    mz.testdrive.run(
+        input=dedent(
+            f"""
+            > SELECT COUNT(*) {comparison_operator} {reached_index} FROM {cluster_setup.get_counter_source_name()};
+            true
+
+            > SELECT COUNT(*) {comparison_operator} {reached_index} FROM {cluster_setup.get_mv_name()};
+            true
+            """
+        ),
+        default_timeout=f"{timeout_in_sec}s",
+        no_reset=True,
+    )
+
+
+def get_current_counter_index(
+    mz: MaterializeApplication, cluster_setup: ClusterSetup
+) -> int:
+    """
+    This query has no timeout. Only use it if is expected to deliver.
+    """
+    reached_value: int = mz.environmentd.sql_query(
+        f"SELECT COUNT(*) FROM {cluster_setup.get_counter_source_name()}"
+    )[0][0]
+    return reached_value
+
+
+def suspend_first_node_of_pod(mz: MaterializeApplication, cluster_name: str) -> None:
```

Comment:
All of your clusters are singe-node, so this can be simplified to `suspend_pod_node`

Reply:
No, `test_replicated_compute_cluster_on_failing_node` is not.